### PR TITLE
DW-568: Added support for inheriting values without creating a submission

### PR DESCRIPTION
**Pull request by rimi-itk**
Sometimes we don't want to create a submission when inheriting form element values, but rather simply pre-populate the form. This change makes this possible.

Screenshot 2022-08-19 at 12 42 53

### CODE REVIEW

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php [hunk 1]
```diff
@@ -164,4 +183,43 @@ public function getExecutableForm($modal, MaestroExecuteInteractive $parent) {
...
+   */
+  private static function setTaskValues($queueID, $values) {
+    $sessionKey = self::getTaskValuesSessionKey($queueID);
+    \Drupal::request()->getSession()->set($sessionKey, $values);
+  }
+
+  /**
+   * Get task values session key.
+   */
+  private static function getTaskValuesSessionKey(int $queueID) {
```
Comment (stanbellcom) [inline src/Plugin/EngineTasks/MaestroWebformInheritTask.php:221]: i suggest to call it formatTaskValuesSessionKey, as it in fact is formating the provided key, not getting it from elsewhere

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php [hunk 2]
```diff
@@ -164,4 +183,43 @@ public function getExecutableForm($modal, MaestroExecuteInteractive $parent) {
...
+   * Set task values in session.
+   */
+  private static function setTaskValues($queueID, $values) {
+    $sessionKey = self::getTaskValuesSessionKey($queueID);
+    \Drupal::request()->getSession()->set($sessionKey, $values);
+  }
+
+  /**
+   * Get task values session key.
+   */
```
Comment (stanbellcom) [inline src/Plugin/EngineTasks/MaestroWebformInheritTask.php:220]: param/returns docs missing

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php [hunk 3]
```diff
@@ -164,4 +183,43 @@ public function getExecutableForm($modal, MaestroExecuteInteractive $parent) {
...
+   * Get task values from session.
+   */
+  private static function getTaskValues($queueID) {
+    $sessionKey = self::getTaskValuesSessionKey($queueID);
+    return \Drupal::request()->getSession()->get($sessionKey);
+  }
+
+  /**
+   * Set task values in session.
+   */
```
Comment (stanbellcom) [inline src/Plugin/EngineTasks/MaestroWebformInheritTask.php:212]: param/returns docs missing

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php [hunk 4]
```diff
@@ -164,4 +183,43 @@ public function getExecutableForm($modal, MaestroExecuteInteractive $parent) {
...
+      $elementId = $element['#webform_key'];
+      if (isset($values['data'][$elementId])) {
+        $element['#default_value'] = $values['data'][$elementId];
+      }
+    }
+  }
+
+  /**
+   * Get task values from session.
+   */
```
Comment (stanbellcom) [inline src/Plugin/EngineTasks/MaestroWebformInheritTask.php:204]: param/returns docs missing

**Review by stanbellcom — CHANGES_REQUESTED** [review]

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php (hunk 1 above)
Comment (rimi-itk) [reply src/Plugin/EngineTasks/MaestroWebformInheritTask.php]: Fixed in https://github.com/OS2Forms/os2forms_forloeb/pull/135/commits/e70e74020e620234bdd0d31bd45276a4a8b4991a.

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php (hunk 2 above)
Comment (rimi-itk) [reply src/Plugin/EngineTasks/MaestroWebformInheritTask.php:241]: Fixed in https://github.com/OS2Forms/os2forms_forloeb/pull/135/commits/2832a962a0afb4fad3fa06fe2fef21d0318a98bd.

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php (hunk 3 above)
Comment (rimi-itk) [reply src/Plugin/EngineTasks/MaestroWebformInheritTask.php:227]: Fixed in https://github.com/OS2Forms/os2forms_forloeb/pull/135/commits/2832a962a0afb4fad3fa06fe2fef21d0318a98bd.

File: src/Plugin/EngineTasks/MaestroWebformInheritTask.php (hunk 4 above)
Comment (rimi-itk) [reply src/Plugin/EngineTasks/MaestroWebformInheritTask.php:211]: Fixed in https://github.com/OS2Forms/os2forms_forloeb/pull/135/commits/2832a962a0afb4fad3fa06fe2fef21d0318a98bd.

**Review by stanbellcom — APPROVED** [review]

**Comment by rimi-itk** [comment]
@stanbellcom, after intensive testing I've made a small, but crucial change to the code (https://github.com/OS2Forms/os2forms_forloeb/pull/135/commits/9e86111d5c316bdcbdee332e3427a5e4b2f70040).

**Review by stanbellcom — APPROVED** [review]

**Pull request merged**